src/price/crypto-chart-box.tsx

```diff
@@ -0,0 +1,472 @@
+  },
+  
+  texts: {
+    fontSize: '18px',
+  },
+
+  active: {
+    borderRadius: '4px',
+    boxShadow: '0px 0px 5px 1px #0dc7dd',
+    outlineWidth: 0,
+    border: 'none'
+  },
+
+  totalPrice: {
+    fontSize: '30px',
+    fontWeight: 300,
+    textAlign: 'center',
+    color: '#ffffff'
+  },
+
+  totalPriceText: {
+    fontSize: '14px',
+    marginTop: '5px',
+    textAlign: 'center',
+    color: '#93b0c1'
+  },
+
+  chartCenterContainer: {
+    position: 'absolute',
+    textAlign: 'center',
+    cursor: 'default',
+    top: '50%',
+    left: '50%',
+    transform: 'translate(-50%, -50%)'
+  },
+
+  chart: {
+    position: 'relative'
+  },
+
+  gearButton: {
+    cursor: 'pointer',
+    display: 'flex',
+    background: 'none',
+    border: 'none',
+    outline: 'none',
+    float: 'right'
+  },
+
+  buttonViewMore: {
+    outline: 'none',
+    fontSize: '12px',
+    color: '#93b0c1',
+    cursor: 'pointer',
+    textTransform: 'uppercase'
+  },
+
+  buttonViewMoreText: {
+    borderBottom: '1px dashed #93b0c1'
+  },
+
+  expandMore: {
+    verticalAlign: 'middle !important'
+  },
+
+  header: {
+    margin: '10px'
+  },
+
+  headerText: {
+    fontSize: '20px',
+    lineHeight: 1.8,
+    opacity: 0.83
+  }
+};
+
+export type Token = {
+  name: string,
+  symbol: string,
+  balance: number,
+  balanceInFiat: number
+}
+
+export type CryptoChartBoxProps = {
+  locale: string,
+  fiatCurrency: string,
+  tokens: Array<Token>,
+  manageCryptoAction?: ((event: React.MouseEvent<HTMLElement>) => void) 
+}
+
+export type StyledProps = WithStyles<keyof typeof styles> & CryptoChartBoxProps;
+
+export type Active = {
+  active: boolean;
+};
+
+export type CryptoChartBoxState = {
+  activations: Array<Active>,
+  displayedTokens: Array<Token>,
+  viewAll: boolean
+};
+
+export type ChartType = {
+  setSelection: Function
+}
+
+export type ChartElementType = {
+  wrapper: ChartWrapperType
+}
+
+export type ChartWrapperType = {
+  getChart: Function
+}
+
+export class CryptoChartBoxComponent extends React.Component<StyledProps, CryptoChartBoxState> {
+  
+  TOP_TOKEN_LIST_SIZE = 5;
+  OTHERS_COLOR = '#71a6b8';
+  otherTokens: Array<Token> = [];
+
+  activations: Array<Active> = [];
+  topTokens: Array<Token> = [];
+  othersToken: Token = {
+    name: 'Others',
+    symbol: 'OTHERS',
+    balance: 0,
+    balanceInFiat: 0
+  };
+
+  state = {
+    activations: this.activations,
+    displayedTokens: this.topTokens,
+    viewAll: true
+  }
+
+  selection = [];
+  chart: ChartType;
+
+  public refs: {
+    pieChart: HTMLElement & ChartElementType;
+  };
+  
+  constructor(props: StyledProps) {
+    super(props);
+
+    this.initState(props);
+    this.initSelection();
+  }
+
+  initState(props: StyledProps) {
+    if (!props.tokens) {
+      return;
+    }
+    props.tokens.forEach((token, index) => {
+      this.state.activations[index] = {active: false};
+    });
+
+    this.topTokens = props.tokens.slice(0, this.TOP_TOKEN_LIST_SIZE);
```

Comment:
This state should be managed in Redux.

What happens to this component if the user's token balance changes while they've got this screen open?

This data is only populated once, when the component loads. The top tokens may change in response to the user sending or receiving tokens and Redux has no way to update this component when that happens.